sway/desktop/transaction.c

```diff
@@ -533,20 +527,31 @@ void transaction_commit_dirty(void) {
 	if (!server.dirty_nodes->length) {
 		return;
 	}
-	struct sway_transaction *transaction = transaction_create();
-	if (!transaction) {
-		return;
+
+	struct sway_transaction *transaction = NULL;
+	if (server.transactions->length > 1) {
```

Comment:
Hmm, I get that doing things like this gets to reuse the dequeuing code, but rather than an array of length 2, I think this would be more descriptive as two separate fields: `server.current_transaction` and `server.pending_transaction` (or whatever naming matches what we do for double-buffering state elsewhere).

Reply:
Agreed, I initially kept the the same just for change simplicity. Updated.